### PR TITLE
resumable

### PR DESCRIPTION
**Pull request by stefanpenner**
\w @lynchbomb 

- [x] cleanup
- [x] discuss
- [ ] more tests 
   hook methods and such should all work as expected, but more discussion is required.

### CODE REVIEW

**Comment by mixonic** [comment]
Neat! How would this interact with Ember, for example in the common case of a budget?

afaict (but I would like to see your version) `mustYield` would close over a value that defines the stop time for the budget and only return `true` when that time is exceeded. However the stop time would need to be recalculated at the start of every runloop, and I'm not clear on what mechanism one would use to achieve that.

Additionally, I believe there should be a mechanism in here that forces a complete sync flush when a new runloop begins.

**Comment by stefanpenner** [comment]
> Neat! How would this interact with Ember, for example in the common case of a budget?

Two things:

* I believe [scheduleIterable](https://github.com/BackburnerJS/backburner.js/pull/204) to be a better approach for budgetary concerns such as chunked rendering. As the "Boss" is glimmer, BB should merely facilitate, as bb really doesn't have enough context on its own for safe pause/resume/cancel. @krisselden / @lynchbomb are exploring this further I believe.
* For what remains, BB pause/resume could be used, maybe when all else fails or for interesting stepwise debugging or...

> afaict (but I would like to see your version) mustYield would close over a value that defines the stop time for the budget and only return true when that time is exceeded. However the stop time would need to be recalculated at the start of every runloop, and I'm not clear on what mechanism one would use to achieve that.

`mustYield` is a placeholder for something like what you describe. My goal here was to ensure the system could be safely paused/resumed while maintaining all existing ordering/performance semantics, so that more exploration with "budgeting" can be done.

In other words, `mustYield` should likely be replaced with something that has roughly similar semantics. (Something the queue flush can consult on a per iteration basis to see if it should proceed or not).

I hope this PR enables further exploration.

File: lib/index.ts [hunk 1]
```diff
@@ -691,9 +707,9 @@ export default class Backburner {
   private _ensureInstance(): DeferredActionQueues {
     let currentInstance = this.currentInstance;
     if (!currentInstance) {
-      const setTimeout = this._platform.setTimeout;
+      const next = this._platform.next || this._platform.setTimeout // TODO: remove the fallback;
```
Comment (krisselden) [inline lib/index.ts]: resolve next at construction
